apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.GenerateClient/Program.cs

```diff
@@ -0,0 +1,286 @@
+            var apiClientName = args[1];
+            var userClientName = args[2];
+
+            var workspace = MSBuildWorkspace.Create(new Dictionary<string, string> { ["TargetFramework"] = "net45" });
+
+            Project project;
+            try
+            {
+                project = await workspace.OpenProjectAsync(projectFile);
+            }
+            catch (FileNotFoundException)
+            {
+                Console.WriteLine($"Could not find project file {projectFile}");
+                return 2;
+            }
+
+            var compilation = await project.GetCompilationAsync();
+            var symbols = compilation.GetSymbolsWithName(name => name == apiClientName, SymbolFilter.Type).ToList();
+            if (symbols.Count != 1)
+            {
+                Console.WriteLine($"Could not find type {apiClientName}");
+                return 2;
+            }
+
+            var apiClientSymbol = (INamedTypeSymbol)symbols[0];
+
+            // Find the API client from a generated file (there should be only one).
+            var generatedApiClientSyntax = apiClientSymbol.DeclaringSyntaxReferences
+                .Select(syntaxReference => syntaxReference.GetSyntax())
+                .Cast<ClassDeclarationSyntax>()
+                .SingleOrDefault(
+                    syntax => syntax.SyntaxTree.GetRoot().GetLeadingTrivia().ToFullString().Contains("// Generated code. DO NOT EDIT!"));
+
+            if (generatedApiClientSyntax == null)
+            {
+                Console.WriteLine($"Could not find an auto-generated file containing the {apiClientName} definition");
+                return 2;
+            }
+
+            var syntaxTree = generatedApiClientSyntax.SyntaxTree;
+            var semanticModel = compilation.GetSemanticModel(syntaxTree);
+            var generator = SyntaxGenerator.GetGenerator(project.GetDocument(syntaxTree));
+            var requestMethodRewriter = new RequestMethodRewriter(semanticModel);
+            var requestMethodImplRewriter = new RequestMethodToImplRewriter();
+
+            var userClientSyntax =
+                (ClassDeclarationSyntax)generator.ClassDeclaration(
+                    userClientName,
+                    accessibility: Accessibility.Public,
+                    modifiers: DeclarationModifiers.Partial)
+                    .WithLeadingTrivia(generatedApiClientSyntax.GetLeadingTrivia());
+            var userClientImplSyntax =
+                (ClassDeclarationSyntax)generator.ClassDeclaration(
+                    userClientName + "Impl",
+                    accessibility: Accessibility.Public,
+                    modifiers: DeclarationModifiers.Sealed | DeclarationModifiers.Partial,
+                    baseType: ParseTypeName(userClientName))
+                    .WithLeadingTrivia(generatedApiClientSyntax.GetLeadingTrivia());
+
+            foreach (var methodSyntax in generatedApiClientSyntax.Members.OfType<MethodDeclarationSyntax>())
```

Comment:
Some comments on this bit could be useful. I'm surprised we're only copying the methods with request objects. Maybe more will come in later commits?

Reply:
Done. I would like to generate more, but no plans at the moment. There was a happy coincidence here where I had already copied the flattened methods so I could provide overloads with a `BigtableByteString` parameter in place of the `ByteString` row keys. Those just now delegate to these copied methods instead of the API client's methods.

Reply:
(now that I have this generator, I would like to auto-generate those copies as well, but it isn't needed for this refactoring)

Reply:
Ah, so the partial class that used to target the "low-level" API now just targets the new API?
Presumably that means we don't have any overloads accepting ByteString, and we're okay with that?

Reply:
Correct.

Technically yes, but since there's an implicit conversion from `ByteString` to `BigtableByteString`, no one should notice the change if they were specifying `ByteString` directly.